jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathSpec.java

```diff
@@ -85,11 +93,165 @@
      */
     String getSuffix();
 
+    /**
+     *  Test if a string is equivalent to this pathspec
+     * @param pathSpec String representation to be normalized and then compared to the {@link #getDeclaration()} string.
+     * @return true if the string, after normalization, represents the same pathspec
+     */
+    boolean is(String pathSpec);
+
     /**
      * Test to see if the provided path matches this path spec
      *
      * @param path the path to test
      * @return true if the path matches this path spec, false otherwise
      */
     boolean matches(String path);
+
+    /**
+     * A Mapping of one or more PathSpecs
+     */
+    class Mapping
+    {
+        private PathSpec[] _pathSpecs = {}; // never null
+
+        /**
+         * @param pathSpec The pathSpec to set, which is assumed to be a {@link ServletPathSpec}
+         */
+        public void addPathSpec(String pathSpec)
+        {
+            Objects.requireNonNull(pathSpec);
+            addPathSpec(new ServletPathSpec(pathSpec));
+        }
+
+        /**
+         * @param pathSpec The pathSpec to set, which may be an arbitrary type of pathspec
+         */
+        public void addPathSpec(PathSpec pathSpec)
+        {
+            Objects.requireNonNull(pathSpec);
+            _pathSpecs = ArrayUtil.addToArray(_pathSpecs, pathSpec, PathSpec.class);
+        }
+
+        /**
+         * Test if the list of path specs contains a particular one.
+         *
+         * @param pathSpec the path spec
```

Comment:
Is this an arbitrary type of path spec, or assumed to be a ServletPathSpec? It would be helpful to keep the language the same, as all the different path specs get a bit confusing.